### PR TITLE
tests: show why sbuild failed

### PR DESCRIPTION
**Pull request by zyga**
The sbuild test seems to fail sometimes, but not due to timeout that
kills the test. When that happens test logs contain absolutely nothing
useful.

Maciej suggested that we can look at the tail of the sbuild logs to do
so. In a practical run invoked with --shell-after there are two log
files:

    snapd_2.40-2_amd64-2019-08-13T09:46:14Z.build
    snapd_2.40-2_amd64.build -> snapd_2.40-2_amd64-2019-08-13T09:46:14Z.build

I don't want to be in the business of extracting that version so I just
take the one with the Z.build suffix.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

File: tests/main/sbuild/task.yaml [hunk 1]
```diff
@@ -29,6 +29,10 @@ restore: |
     rm -f /etc/schroot/chroot.d/sid-amd64-sbuild-*
 
 debug: |
+    # There there's a log file and a symbolic link pointing to it.
+    # The non-symlink has a time-stamp and we can match on the "Z" timezone
+    # marker to find it.
+    tail -n 100 *Z.build
```
Comment (chipaca) [inline tests/main/sbuild/task.yaml]: and if there isn't a log file?

Comment (zyga) [reply tests/main/sbuild/task.yaml]: Then the test will fail anyway.

Comment (chipaca) [reply tests/main/sbuild/task.yaml]: yes, but so will the debug, and then you can't `-debug`?

Comment (bboozzoo) [inline tests/main/sbuild/task.yaml]:
```suggestion
    test "$(find . -maxdepth 1 -name '*Z.build' | wc -l)" -ge 1 &&  tail -n 100 *Z.build
```

Comment (zyga) [reply tests/main/sbuild/task.yaml]: Do you mean that this would prevent a `-debug` shell?

File: tests/main/sbuild/task.yaml [hunk 2]
```diff
@@ -29,6 +29,10 @@ restore: |
     rm -f /etc/schroot/chroot.d/sid-amd64-sbuild-*
 
 debug: |
+    # There there's a log file and a symbolic link pointing to it.
+    # The non-symlink has a time-stamp and we can match on the "Z" timezone
+    # marker to find it.
+    test "$(find . -maxdepth 1 -name '*Z.build' | wc -l)" -ge 1 &&  tail -n 100 *Z.build
```
Comment (zyga) [inline tests/main/sbuild/task.yaml]:
```suggestion
    test "$(find . -maxdepth 1 -name '*Z.build' | wc -l)" -ge 1 &&  tail -n 100 ./*Z.build
```

File: tests/main/sbuild/task.yaml [hunk 3]
```diff
@@ -29,6 +29,10 @@ restore: |
     rm -f /etc/schroot/chroot.d/sid-amd64-sbuild-*
 
 debug: |
+    # There there's a log file and a symbolic link pointing to it.
```
Comment (anonymouse64) [inline tests/main/sbuild/task.yaml]:
```suggestion
    # Test that there's a log file and a symbolic link pointing to it.
```

**Pull request merged**